pkg/vsphere/vm/vm.go

```diff
@@ -52,6 +53,9 @@ type VirtualMachine struct {
 	*object.VirtualMachine
 
 	*session.Session
+
+	// Fixing is true means the VM will be unregistered and registered back.
+	Fixing *atomic.Value
```

Comment:
for clarity if I was doing that I would use an integer value for this flag.
atomic.Value is a lot of overhead for a simple flag.

```
// supposed to be atomic.
	var fixFlag int32
	
	v := atomic.LoadInt32(&fixFlag)
	atomic.StoreInt32(&fixFlag, 1)
```